modules/influxdb/influxdb.go

```diff
@@ -57,39 +73,23 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 			// "Open shard" which is the last thing that happens before the server is ready to accept connections.
 			// This is probably different for InfluxDB 2.x, but that is left as an exercise for the reader.
 			strategies := []wait.Strategy{
-				genericContainerReq.WaitingFor,
+				wait.ForListeningPort("8086/tcp"),
 				wait.ForLog("influxdb init process in progress..."),
 				wait.ForLog("Server shutdown completed"),
 				wait.ForLog("Opened shard"),
 			}
-			genericContainerReq.WaitingFor = wait.ForAll(strategies...)
-			hasInitDb = true
-			break
+			return wait.ForAll(strategies...)
```

Comment:
suggestion: Testing this locally we should be able to use the http health check, but it would need to come after the first shutdown otherwise it could pass prematurely.

So the overall flow seems like it should be:
1. If init DB wait for DB init and shutdown
2. Wait for http health check to succeed.

Note the waits for 1 and 2 are combined, as well as any custom wait set by the user.

Reply:
Alright, I’ll stick with wait.ForLog("Server shutdown completed") and then wait for the HTTP health check to pass.